core/src/main/java/org/infinispan/remoting/inboundhandler/action/ActionState.java

```diff
@@ -14,20 +14,20 @@
  */
 public class ActionState {
 
-
-   private final RemoteLockCommand command;
+   private final ReplicableCommand command;
    private final int commandTopologyId;
    private volatile long timeout;
    private volatile List<Object> filteredKeys;
 
-   public ActionState(RemoteLockCommand command, int commandTopologyId, long timeout) {
+   public ActionState(ReplicableCommand command, int commandTopologyId, long timeout) {
       this.command = command;
       this.commandTopologyId = commandTopologyId;
       this.timeout = timeout;
    }
 
-   public final RemoteLockCommand getCommand() {
-      return command;
+   public final <T extends ReplicableCommand> T getCommand() {
+      //noinspection unchecked
+      return (T) command;
```

Comment:
Same here, you wouldn't need the generic parameter if `RemoteLockCommand` extended `ReplicableCommand`.